### PR TITLE
Implement `tensor.special.logit` helper

### LINKED ISSUE #563: Add tensor helper equivalent for `scipy.special.logit`
### Description

Currently missing: https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.logit.html#scipy.special.logit

Should not require a new Op, a simple function with existing PyTensor ops would suffiec

### PR DESCRIPTION
**Pull request by HarshvirSandhu**
## Description
Implement tensor helper equivalent for [scipy.special.logit](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.logit.html#scipy.special.logit)

## Related Issue

- [x] Closes #563  

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)

## Type of change

- [x] New feature / enhancement

### CODE REVIEW

File: tests/tensor/test_special.py [hunk 1]
```diff
@@ -206,6 +208,18 @@ def test_factorial(n):
...
+def test_logit():
+    _x = vector("x")
+    actual_fn = function([_x], logit(_x))
+
+    x = random_ranged(0, 1, (5,))
+    actual = actual_fn(x)
+    expected = scipy_logit(x)
+    np.testing.assert_allclose(
+        actual, expected, rtol=1e-7 if config.floatX == "float64" else 1e-5
+    )
```
Comment (ricardoV94) [inline tests/tensor/test_special.py:220]: Perhaps a bit more readable?

```suggestion
    x = vector("x")
    actual_fn = function([x], logit(x))

    x_test = np.linspace(0, 1)
    actual = actual_fn(x_test)
    expected = scipy_logit(x_test)
    np.testing.assert_allclose(
        actual, expected, rtol=1e-7 if config.floatX == "float64" else 1e-5
    )
```

**Comment by ricardoV94** [comment]
Looks pretty good, left just a cosmetic suggestion to the test

**Comment by ricardoV94** [comment]
Pre-commit is failing though

**Review by ricardoV94 — APPROVED** [review]
Thanks !

**Comment by HarshvirSandhu** [comment]
@ricardoV94 
It looks like a test is failing on Python 3.12
But the test is not failing on my local computer.

**Comment by ricardoV94** [comment]
> @ricardoV94 It looks like a test is failing on Python 3.12 But the test is not failing on my local computer.

Test is failing on float32, you can reproduce by adding this snippet at the top of the test file. Probably just means you need to cast `x_test` to `pytensor.config.floatX` in the test (this happens all the time):

```python
import pytensor
pytensor.config.floatX = "float32"
```

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
Thanks a lot @HarshvirSandhu

**Pull request merged**